File: api/src/test/java/net/blueshell/api/factory/dto/AddressDTOFactory.java
package net.blueshell.api.factory.dto;

import lombok.RequiredArgsConstructor;
import net.blueshell.api.dto.AddressDTO;
import org.springframework.stereotype.Component;

/**
 * Factory for AddressDTO test instances.
 */
@Component
@RequiredArgsConstructor
public class AddressDTOFactory extends BaseDtoFactory<AddressDTO> {

    @Override
    public Class<AddressDTO> targetType() {
        return AddressDTO.class;
    }

    @Override
    public AddressDTO createBasic() {
        AddressDTO dto = new AddressDTO();
        dto.setCountry("NL");
        dto.setCity("Enschede");
        dto.setStreet("Test Street");
        dto.setHouseNumber("123");
        dto.setZipCode("1234 AB");
        dto.setCreatedAt(now());
        return dto;
    }
}

File: api/src/test/kotlin/net/blueshell/api/factory/dto/AddressDTOFactory.kt
package net.blueshell.api.factory.dto

import net.blueshell.api.feature.user.dto.AddressDTO
import org.springframework.stereotype.Component

/**
 * Factory for AddressDTO test instances.
 */
@Component
class AddressDTOFactory : BaseDtoFactory<AddressDTO>() {

    override fun targetType(): Class<AddressDTO> = AddressDTO::class.java

    override fun createBasic(): AddressDTO {
        val dto = AddressDTO()
        dto.country = "NL"
        dto.city = "Enschede"
        dto.street = "Test Street"
        dto.houseNumber = "123"
        dto.zipCode = "1234 AB"
        dto.createdAt = now()
        return dto
    }
}
